src/jupyterlabcontroller/constants.py

```diff
@@ -3,20 +3,42 @@
 from datetime import timedelta
 from pathlib import Path
 
+__all__ = [
+    "CONFIGURATION_PATH",
+    "DOCKER_SECRETS_PATH",
+    "DROPDOWN_SENTINEL_VALUE",
+    "GROUPNAME_REGEX",
+    "IMAGE_REFRESH_INTERVAL",
+    "KUBERNETES_REQUEST_TIMEOUT",
+    "LAB_STATE_REFRESH_INTERVAL",
+    "LIMIT_TO_REQUEST_RATIO",
+    "METADATA_PATH",
+    "SPAWNER_FORM_TEMPLATE",
+    "USERNAME_REGEX",
+]
+
 CONFIGURATION_PATH = Path("/etc/nublado/config.yaml")
 """Default path to controller configuration."""
 
 DOCKER_SECRETS_PATH = Path("/etc/secrets/.dockerconfigjson")
 """Default path to the Docker API secrets."""
 
-METADATA_PATH = Path("/etc/podinfo")
-"""Default path to injected pod metadata."""
+DROPDOWN_SENTINEL_VALUE = "use_image_from_dropdown"
+"""Used in the lab form for ``image_list`` when ``image_dropdown`` is used."""
 
 IMAGE_REFRESH_INTERVAL = timedelta(minutes=5)
 """How frequently to refresh the list of remote and cached images."""
 
-DROPDOWN_SENTINEL_VALUE = "use_image_from_dropdown"
-"""Used in the lab form for ``image_list`` when ``image_dropdown`` is used."""
+LAB_STATE_REFRESH_INTERVAL = timedelta(minutes=60)
```

Comment:
That's far too long for a default, I think.  5 minutes, maybe?

Reply:
I'm worried about the overhead here when we have 600 running labs. I'm also changing the code so that when JupyterHub polls the status of a lab, we check Kubernetes directly and detect disappeared lab pods then, so this is just a fallback and cleanup for updating the state of pods we're tracking but that JupyterHub hasn't asked us about, or to fix the state if someone manually deleted the rest of the namespace. For those purposes, I think once an hour should be okay.